tool/lsm.go

```diff
@@ -282,15 +280,6 @@ func (l *lsmT) buildEdits(
 			Added:   make(map[int][]base.FileNum),
 			Deleted: make(map[int][]base.FileNum),
 		}
-		for df := range ve.DeletedFiles {
-			edit.Deleted[df.Level] = append(edit.Deleted[df.Level], df.FileNum)
-			for j, f := range currentFiles[df.Level] {
-				if f.FileNum == df.FileNum {
-					copy(currentFiles[df.Level][j:], currentFiles[df.Level][j+1:])
-					currentFiles[df.Level] = currentFiles[df.Level][:len(currentFiles[df.Level])-1]
-				}
-			}
-		}
```

Comment:
I'm not sure why the `ve.DeletedFiles` iteration was before the `ve.NewFiles` interation (or if it was necessary) but when changing the order the tool seems to function as normal. 

Changing the order helped avoid the logic of prepopulating `currentFiles` to include the BVE's new files (now they are added before `ve.DeletedFiles` are handled).